### PR TITLE
Allow append rows in Streaming workbook

### PR DESCRIPTION
**Pull request by jlolling**
According to the ticket: https://bz.apache.org/bugzilla/show_bug.cgi?id=67646
Here the pull request to solve the issue.

### CODE REVIEW

File: poi-ooxml/src/main/java/org/apache/poi/xssf/streaming/SXSSFSheet.java [hunk 1]
```diff
@@ -1145,7 +1152,7 @@ public boolean isDisplayRowColHeadings() {
      * Breaks occur above the specified row and left of the specified column inclusive.
      *
      * For example, {@code sheet.setColumnBreak(2);} breaks the sheet into two parts
-     * with columns A,B,C in the first and D,E,... in the second. Similar, {@code sheet.setRowBreak(2);}
+     * with columns A,B,C in the first and D,E,... in the second. Simuilar, {@code sheet.setRowBreak(2);}
```
Comment (pjfanning) [inline poi-ooxml/src/main/java/org/apache/poi/xssf/streaming/SXSSFSheet.java:1155]: Can you fix the typo here?

File: poi-ooxml/src/main/java/org/apache/poi/xssf/streaming/SXSSFSheet.java [hunk 2]
```diff
@@ -1240,11 +1247,11 @@ public void setColumnGroupCollapsed(int columnNumber, boolean collapsed) {
      */
     @Override
     public void groupColumn(int fromColumn, int toColumn) {
-        _sh.groupColumn(fromColumn, toColumn);
+        _sh.groupColumn(fromColumn,toColumn);
     }
 
     /**
-     * Ungroup a range of columns that were previously grouped
+     * Ungroup a range of columns that were previously groupped
```
Comment (pjfanning) [inline poi-ooxml/src/main/java/org/apache/poi/xssf/streaming/SXSSFSheet.java:1254]: another typo - please do not give us PRs where you randomly change code unrelated to your fix

File: poi-ooxml/src/main/java/org/apache/poi/xssf/streaming/SXSSFSheet.java [hunk 3]
```diff
@@ -1240,11 +1247,11 @@ public void setColumnGroupCollapsed(int columnNumber, boolean collapsed) {
      */
     @Override
     public void groupColumn(int fromColumn, int toColumn) {
-        _sh.groupColumn(fromColumn, toColumn);
```
Comment (pjfanning) [inline poi-ooxml/src/main/java/org/apache/poi/xssf/streaming/SXSSFSheet.java:1243]: why are you reformatting this code?

File: poi-ooxml/src/main/java/org/apache/poi/xssf/streaming/SXSSFSheet.java [hunk 4]
```diff
@@ -1349,33 +1361,33 @@ public void ungroupRow(int fromRow, int toRow) {
...
+     * @throws RuntimeException if collapse is false as this is not implemented for SXSSF.
      */
     @Override
     public void setRowGroupCollapsed(int row, boolean collapse) {
         if (collapse) {
             collapseRow(row);
         } else {
             //expandRow(rowIndex);
-            throw new IllegalStateException("Unable to expand row: Not Implemented");
+            throw new RuntimeException("Unable to expand row: Not Implemented");
```
Comment (pjfanning) [inline poi-ooxml/src/main/java/org/apache/poi/xssf/streaming/SXSSFSheet.java:1374]: please do not change the exception type

**Review by pjfanning — CHANGES_REQUESTED** [review]
Can you add a test case?

File: poi-ooxml/src/main/java/org/apache/poi/xssf/streaming/SXSSFSheet.java [hunk 5]
```diff
@@ -61,11 +62,7 @@ protected SXSSFSheet(SXSSFWorkbook workbook, XSSFSheet xSheet, int randomAccessW
         _sh = xSheet;
         calculateLeftAndRightMostColumns(xSheet);
         setRandomAccessWindowSize(randomAccessWindowSize);
-        try {
-            _autoSizeColumnTracker = new AutoSizeColumnTracker(this);
-        } catch (UnsatisfiedLinkError | InternalError e) {
-            LOG.atWarn().log("Failed to create AutoSizeColumnTracker, possibly due to fonts not being installed in your OS", e);
-        }
+        _autoSizeColumnTracker = new AutoSizeColumnTracker(this);
```
Comment (pjfanning) [inline poi-ooxml/src/main/java/org/apache/poi/xssf/streaming/SXSSFSheet.java:65]: please do not randomly change code like this- undo this change or the PR is rejected

**Comment by jlolling** [comment]
I will try to add a test case
I will also fix the typos but they are not caused by me.

**Comment by jlolling** [comment]
I have not “randomly” changed code. I have started from a code some weeks earlier. I will rebase my branch and continue.

File: poi-ooxml/src/main/java/org/apache/poi/xssf/streaming/SXSSFSheet.java [hunk 6]
```diff
@@ -191,8 +192,13 @@ public void removeRow(Row row) {
      * @return Row representing the rownumber or null if its not defined on the sheet
      */
     @Override
-    public SXSSFRow getRow(int rownum) {
-        return _rows.get(rownum);
+    public Row getRow(int rownum) {
+    	Row row = _rows.get(rownum);
+    	// jlolling: allow reading all the content
```
Comment (pjfanning) [inline poi-ooxml/src/main/java/org/apache/poi/xssf/streaming/SXSSFSheet.java:197]: In the ASF, we believe in shared ownership of the code. Please do not annotate code with your name.

**Comment by jlolling** [comment]
I will remove my name and set the ticket number instead. OK?
